src/backend/distributed/utils/citus_ruleutils.c

```diff
@@ -209,6 +210,41 @@ AppendOptionListToString(StringInfo stringBuffer, List *optionList)
 
 
 /*
+ * pg_get_sequencedef_string returns the definition of a given sequence. This
+ * definition includes explicit values for all CREATE SEQUENCE options.
+ */
+char *
+pg_get_sequencedef_string(Oid sequenceRelationId)
+{
+	char *qualifiedSequenceName = generate_relation_name(sequenceRelationId, NIL);
+	char *sequenceDef = NULL;
+	Form_pg_sequence pgSequenceForm = NULL;
+
+	Relation sequenceRel = heap_open(sequenceRelationId, AccessShareLock);
```

Comment:
This needs a lock on the pg_class entry first, i.e. something like LockRelationOid().